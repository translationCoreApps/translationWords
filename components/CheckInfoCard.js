//CheckInfoCard.js//
/**
 * @author Ian Hoegen
 * @description This component is a display component for the Check Info Cards.
 */
const React = api.React;
const RB = api.ReactBootstrap;
const {Row, Glyphicon, Col} = RB;
const styles = require('../css/style.js');
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';

class CheckInfoCard extends React.Component {
    /**
     * @description This function removes any non paragraph text from the text, then shortens it
     *              if the text is too long, also adding ellipses.
     * @param {String} text - The TranslationHelps file
     * @return {String} - The text to display
     */
    cleanText(text) {
      let splitLine = text.split('\n');
      let finalArray = []
      for (let i = 0; i < splitLine.length; i++) {
        if (splitLine[i] !== '' && !~splitLine[i].indexOf("#")) {
          finalArray.push(splitLine[i])
        }
      }
      let maxLength = 115;
      let finalString = "";
      let chosenString = finalArray[0];
      let splitString = chosenString.split(' ');
      for (let word of splitString) {
        if ((finalString + ' ' + word).length >= maxLength) {
          finalString+= '...'
          break;
        }
        finalString += ' ';
        finalString += word;
      }
      return finalString;
    }
    render() {
      return (
        <div style={{margin: '10px'}}>
        <Card zDepth={2} style={{ background: '#03A9F4', padding: "20px"}}>
          <Row>
            <Col md={4} style={{borderRight: '1px solid #FFFFFF'}}>
              <div style={styles.title}>
                {this.props.title}
              </div>
            </Col>
            <Col md ={6}>
              <div style={{color: '#FFFFFF'}}>
                {this.cleanText(this.props.file)}
              </div>
            </Col>
            <Col md={2}>
              <div onClick={this.props.showHelps ? null : this.props.openHelps}
              style={this.props.showHelps ? styles.linkInactive : styles.linkActive}>
                See More
              </div>
            </Col>
          </Row>
        </Card>
        </div>
      );
    }
}

module.exports = CheckInfoCard;
