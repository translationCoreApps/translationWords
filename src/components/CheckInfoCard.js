/* eslint-env jest */

/**
 * @file CheckInfoCard.js
 * @description This component is a display component for the Check Info Cards.
 */
import React from 'react';
import styles from '../css/style.js';
import PropTypes from 'prop-types';

class CheckInfoCard extends React.Component {
    /**
     * @description This function removes any non paragraph text from the text, then shortens it
     *              if the text is too long, also adding ellipses.
     * @param {String} text - The TranslationHelps file
     * @return {String} - The text to display
     */
    cleanText(text) {
      let splitLine = text.split('\n');
      if (splitLine.length == 1 && splitLine[0] == "") return "";
      let finalArray = [];
      for (let i = 0; i < splitLine.length; i++) {
        if (splitLine[i] !== '' && !~splitLine[i].indexOf("#")) {
          finalArray.push(splitLine[i]);
        }
      }
      let maxLength = 225;
      let finalString = "";
      let chosenString = finalArray[0];
      let splitString = chosenString.split(' ');
      for (let word of splitString) {
        if ((finalString + ' ' + word).length >= maxLength) {
          finalString+= '...';
          break;
        }
        finalString += ' ';
        finalString += word;
      }
      return finalString;
    }

    render() {
      let cleanText;
      if (this.props.file) cleanText = this.cleanText(this.props.file);

      return (
        <div style={styles.checkInfo}>
            <div style={styles.leftSide}>
                <div style={styles.title}>
                    {this.props.title}
                </div>
            </div>
          <div style={styles.rightSide}>
              <div style={styles.phrase}>
              {cleanText}
            </div>
            <div onClick={this.props.showHelps ? null : this.props.openHelps} style={this.props.showHelps ? styles.linkInactive : styles.linkActive}>
              See More
            </div>
          </div>
        </div>
      );
    }
}

CheckInfoCard.propTypes = {
  file: PropTypes.string,
  title: PropTypes.string.isRequired,
  showHelps: PropTypes.bool,
  openHelps: PropTypes.func
};

module.exports = CheckInfoCard;
