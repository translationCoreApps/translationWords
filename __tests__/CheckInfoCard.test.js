/* eslint-env jest */
import React from 'react';
import CheckInfoCard from '../src/components/CheckInfoCard';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';

describe('CheckInfoCard component Tests', () => {
  it('check CheckInfoCard', () => {
    let props = {
      translate:k=>k,
      file: 'this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. ',
      title: 'title'
    };
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckInfoCard {...props} />
      </MuiThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
