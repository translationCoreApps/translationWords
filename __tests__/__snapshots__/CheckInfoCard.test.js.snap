// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInfoCard component Tests check CheckInfoCard 1`] = `
<div
  style={
    Object {
      "backgroundColor": "var(--accent-color-dark)",
      "borderRadius": "2px",
      "boxShadow": "0 3px 10px var(--background-color)",
      "color": "var(--reverse-color)",
      "display": "flex",
      "flex": "0 0 120px",
      "margin": "0 10px",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "16px",
          "fontWeight": "bold",
          "maxHeight": "100px",
          "overflowY": "auto",
          "padding": "0 20px",
          "textAlign": "center",
        }
      }
    >
      title
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--reverse-color)",
        "display": "flex",
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "maxHeight": "80px",
          "overflowY": "auto",
          "padding": "0 20px",
          "textAlign": "center",
        }
      }
    >
       this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. this is a test of the check info card component. this is a test of the check...
    </div>
    <div
      onClick={undefined}
      style={
        Object {
          "cursor": "pointer",
          "display": "flex",
          "fontWeight": "bold",
          "justifyContent": "flex-end",
          "margin": "5px 20px",
        }
      }
    >
      see_more
    </div>
  </div>
</div>
`;
